src/Tests/Tests/Test_FastConv.cpp

```diff
@@ -23,19 +25,158 @@ namespace fastconv_test {
     protected:
         void SetUp() override
         {
+            m_pCFastConv = new CFastConv();
+
+            m_IrLength = 51;
+            m_ImpulseLength = 10;
+            m_Ir = new float[m_IrLength];
+            m_Impulse = new float[m_ImpulseLength];
+
+            for (int i = 0; i < m_IrLength; i++)
+            {
+                if (i < m_ImpulseLength) { m_Impulse[i] = 0; };
+                m_Ir[i] = (rand() / static_cast<float>(RAND_MAX));
```

Comment:
This may only be giving positive values so you will have to scale it by a factor of RAND_MAX/2 and then subtract 1 to get values between -1 and 1 (like in an audio context!)